src/axelar/AxelarTransceiver.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0 <0.9.0;
+
+import "wormhole-solidity-sdk/Utils.sol";
+
+import { TransceiverStructs } from "@wormhole-foundation/native_token_transfer/libraries/TransceiverStructs.sol";
+import { INttManager } from "@wormhole-foundation/native_token_transfer/interfaces/INttManager.sol";
+
+import {AxelarExecutable} from "@axelar-network/axelar-gmp-sdk-solidity/contracts/executable/AxelarExecutable.sol";
+import {IAxelarGasService} from "@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IAxelarGasService.sol";
+import {
+    StringToAddress, AddressToString
+} from "@axelar-network/axelar-gmp-sdk-solidity/contracts/libs/AddressString.sol";
+// Use their libraries
+import { Transceiver } from "@wormhole-foundation/native_token_transfer/Transceiver/Transceiver.sol";
+
+import { IAxelarTransceiver } from './interfaces/IAxelarTransceiver.sol';
+
+contract AxelarTransceiver is IAxelarTransceiver, AxelarExecutable, Transceiver {
+    IAxelarGasService public immutable gasService;
+
+    // These mappings are used to convert between chainId and chainName as Axelar accept chainName as string format
+    struct AxelarTransceiverStorage {
+        mapping(uint16 => string) idToAxelarChainIds;
+        mapping(string => uint16) axelarChainIdToId;
+        mapping(uint16 => string) idToAxelarAddress;
+        mapping(string => uint16) axelarAddressToId;
+    }
+
+    // keccak256('AxelarTransceiver.Slot') - 1
+    bytes32 internal constant AXELAR_TRANSCEIVER_STORAGE_SLOT = 0x16cc7b9f29b247db6f6c7350203c763b8802896c91208a944bb4707de3f359a6;
+
+    error UnsupportedMessageType();
+    error InvalidSibling(uint16 chainId, string sourceChain, string sourceAddress);
+    error NotImplemented();
+
+    constructor(address _gateway, address _gasService, address _manager)
+        AxelarExecutable(_gateway)
+        Transceiver(_manager)
+    {
+        gasService = IAxelarGasService(_gasService);
+    }
+
+    function _setup(bytes calldata params) internal override {
+    }
+
+    function contractId() external pure override returns (bytes32) {
+        return keccak256('axelar-transceiver');
+    }
+
+    /**
+     * Set the bridge manager contract address
+     * @param chainId The chainId of the chain. This is used to identify the chain in the EndpointManager.
+     * @param chainName The chainName of the chain. This is used to identify the chain in the AxelarGateway.
+     */
+    function setAxelarChainId(uint16 chainId, string calldata chainName, string calldata axelarAddress) external onlyOwner {
+        AxelarTransceiverStorage storage storage_ = _storage();
+        storage_.idToAxelarChainIds[chainId] = chainName;
```

Comment:
should we allow overriding already set addresses? maybe that should require going through an upgrade for additional trust. does Wormhole follow a certain pattern here?

Reply:
The Wormhole Transceiver trusts the admin of the contract. So, it does allow overriding already set addresses.
The NTT Manager also trusts the admin of the contract, so it allows overriding already set manager peers on other chains